music-playlist-creator/script.js

```diff
@@ -13,4 +36,104 @@ window.onclick = function(event) { //also attached the onclick handler to the wi
+
+
+//function will make a card using JS
+const createPlaylistCard = function(card){
+
+  const playlistArticle = document.createElement('article'); //creates an <article> element
+  playlistArticle.className = 'card'; //allows us to give the article its proper class
+
+  playlistArticle.addEventListener("click", () => { //want to pass in the card data when we open the modal so that we can populate modal information
+    openModal(card);
+  })
+
+  //below just made one fat template literal work smarter not harder. It is the exact same html I had except for the insertion of the variables from our card object which contains playlist details
+
+  //we alter the innerHTML of the article we created above by just adding the literal
+  playlistArticle.innerHTML += `
+    <img class = "playlist-image" src="${card.playlist_art}" alt="Playlist image placeholder">
+
+    <div class = "card-content-padding">
+      <h2>${card.playlist_name}</h2>
+      <p>${card.playlist_author}</p>
+        <div class = "likes-incard">
+          <img class = "like-img" src="assets/img/unlikedHeart.svg" alt="Unliked heart image" width = 35px height = 35px>
+          <p class = "like-count">${card.like_count}</p>
+        </div>
+    </div>
+  `
+
+  //finally we want to append this playlistArticle to the playlistCards section we selected at the top of our code.
+  playlistCards.appendChild(playlistArticle);
+
+  /* ***BELOW CODE IS GOING TO BE FOR ALL LIKE BUTTON FUNCTIONALITY *** */
+
+  const likeImg = playlistArticle.querySelector(".like-img"); //fetch the likeImg tag so that we can change the icon when user clicks it
+  const likeCount = playlistArticle.querySelector(".like-count"); //fetch the p tag that contains the like count because we need to update it when user likes song
+
+  //let us add an event listener to the likeImg which is the heart. Whenever it is clicked we want to toggle the kind of heart that is being displayed. We are attaching listener inside this loop because if we do it outside the playlist card html might not even be rendered yet.
+  likeImg.addEventListener("click", (event)=>{
+
+    event.stopPropagation(); // prevent modal open. Event propogation stops bubbling which is when an event gets sent up to the parent and triggers it too aka the modal open.
+    if(likeImg.src.includes("unlikedHeart")){ //if it is not liked
+      likeImg.src = "assets/img/likedHeart.svg"; //changed it to liked logo
+      card.like_count++; //update the like count
+      likeCount.textContent = card.like_count; //and change the textContent of likeCount p tag since it increased
+    }
+
+    else{ //if it is liked
+      likeImg.src = "assets/img/unlikedHeart.svg"; //change icon to unliked
+      card.like_count--; //decrease the like count
+      likeCount.textContent = card.like_count; ///and update the p tag accordingly
+    }
+
+
```

Comment:
Minor note: I've seen many empty lines in between, at some point they could be too many lines making it difficult to scroll through the code. There are multiple standards in the industry, a popular one on Javascript is the Airbnb one for example [[ref](https://github.com/airbnb/javascript?tab=readme-ov-file#whitespace)]. You can read it in the future if you're interested, but usually we use linters that automatically format the code with that standards making the code consistent and easy to follow for the engineers.